packages/core/contracts-ovm/insured-bridge/implementation/BridgePoolFactory.sol

```diff
@@ -0,0 +1,189 @@
+     * @param _l2Gas Gas limit to set for relayed message on L2
+     */
+    function whitelistToken(
+        address _l1Token,
+        address _l2Token,
+        address _bridgePool,
+        uint32 _l2Gas
+    ) public onlyOwner {
+        require(_getCollateralWhitelist().isOnWhitelist(address(_l1Token)), "Payment token not whitelisted");
+        // We want to prevent any situation where a token mapping is whitelisted on this contract but not on the
+        // corresponding L2 contract.
+        require(depositContract != address(0), "Deposit contract not set");
+
+        L1TokenRelationships storage whitelistedToken = whitelistedTokens[_l1Token];
+        whitelistedToken.l2Token = _l2Token;
+        sendCrossDomainMessage(
+            depositContract,
+            _l2Gas,
+            abi.encodeWithSignature("whitelistToken(address,address)", _l1Token, whitelistedToken.l2Token)
+        );
+
+        // TODO: This contract should deploy a new BridgePool if the address is set to 0x0 at this point.
+        whitelistedToken.bridgePool = _bridgePool;
+
+        emit WhitelistToken(_l1Token, whitelistedToken.l2Token, whitelistedToken.bridgePool);
+    }
+
+    function pauseL2Deposits() public onlyOwner {}
+
+    // Interface view methods exposed to child BridgePool contracts.
+    function getFinder() external view override returns (address) {
+        return finder;
+    }
+
+    function getDepositContract() external view override returns (address) {
+        return depositContract;
+    }
+
+    function getWhitelistedToken(address l1Token) external view override returns (L1TokenRelationships memory) {
+        return whitelistedTokens[l1Token];
+    }
+
+    function getOptimisticOracleLiveness() external view override returns (uint256) {
+        return optimisticOracleLiveness;
+    }
+
+    function getProposerBondPct() external view override returns (uint256) {
+        return proposerBondPct;
+    }
+
+    function getIdentifier() external view override returns (bytes32) {
+        return identifier;
+    }
+
+    // Internal functions
+
+    function _getIdentifierWhitelist() private view returns (IdentifierWhitelistInterface) {
+        return
+            IdentifierWhitelistInterface(
+                FinderInterface(finder).getImplementationAddress(OracleInterfaces.IdentifierWhitelist)
+            );
+    }
+
+    function _getCollateralWhitelist() private view returns (AddressWhitelistInterface) {
+        return
+            AddressWhitelistInterface(
+                FinderInterface(finder).getImplementationAddress(OracleInterfaces.CollateralWhitelist)
+            );
+    }
+
+    function _setIdentifier(bytes32 _identifier) private {
+        require(_getIdentifierWhitelist().isIdentifierSupported(_identifier), "Identifier not registered");
+        // TODO: Should we validate this _identifier? Perhaps check that its not 0x?
```

Comment:
yup. should defo not be able to set it to blank

Reply:
But then again, we check that the identifier is on the Identifier whitelist. So anything that's whitelisted is presumably already sanity checked?

Reply:
that's a great point. if it's on the whitelist. I it's a fine identifier